_posts/2019-08-08-hacking-for-cruise-control.md

```diff
@@ -0,0 +1,227 @@
+---
+layout: post
+title:  "Hacking Strimzi for Cruise Control"
+date: 2019-08-08
+author: kyle_liberti
+---
+
+Cruise Control is a higly customizable Kafka partition balancing and monitoring software run in production at Linkedin. Cruise Controls’s component pluggability makes it flexible to customize for different environments. This makes it possible to extend Cruise Control to work with Strimzi in an Kubernetes environment with a reasonable amount of effort. This post will walk through how to get Cruise Control working with a Stimzi deployed Kafka cluster.
+
+# Prequisite Outline
+
+Like any proof of concept, there are a few actions to consider when putting it together. For getting Cruise Control to work with Strimzi these actions are: 
+
+- Aligning Kafka versions
+- Gathering Kafka Metrics
+- Accessing Zookeeper endpoint
+- Containerizing and deploying Cruise Control
+
+# Aligning Kafka versions
+
+Its worth noting that the latest Kafka version which Cruise Control supports at this time is 2.0.1. The latest version of Strimzi which supports Kafka version 2.0.1 is 0.11.4. Although its possible to run Cruise Control alongside with later versions of Kafka, 2.0.1+, it is not supported. Keep this in mind when deciding which version of Strimzi to deploy. This post will stick to using Strimzi 0.11.4, but feel free to experiment with more updated versions.
+
+# Gathering Metrics
+
+In order to estimate partition activity and resources, Cruise Control needs a steady source of metric data from Kafka. Fortunately, Kafka tracks all sorts of metrics and makes them all accessible through pluggable components know as `metric reporters`. Kafka metric reporters designate what, where, when, and how Kafka metrics are exported. For instance, by default Kafka uses a JMX metric reporter to expose all of its metrics through JMX, by wrapping Kafka metric objects into MBeans. Upon creation of a new Kafka metric object, Kafka will put the metric object in a concurrent hash map for future tracking and pass the metric object to each running metric reporter for its own filtering, formating, and storage. Afterwards, the metric reporters will monitor the metric objects for Kafka stat updates.
```

Comment:
Backticks (and these are backticks, right?) in markdown will be rendered as monospace/code. Probably safer to just use apostrophe.

```suggestion
In order to estimate partition activity and resources, Cruise Control needs a steady source of metric data from Kafka. Fortunately, Kafka tracks all sorts of metrics and makes them all accessible through pluggable components know as 'metric reporters'. Kafka metric reporters designate what, where, when, and how Kafka metrics are exported. For instance, by default Kafka uses a JMX metric reporter to expose all of its metrics through JMX, by wrapping Kafka metric objects into MBeans. Upon creation of a new Kafka metric object, Kafka will put the metric object in a concurrent hash map for future tracking and pass the metric object to each running metric reporter for its own filtering, formating, and storage. Afterwards, the metric reporters will monitor the metric objects for Kafka stat updates.
```